hops/experiment_impl/util/experiment_utils.py

```diff
@@ -222,18 +222,19 @@ def _cleanup(tensorboard, gpu_thread):
     finally:
         tensorboard._reset_global()
 
-    # Close and logging fd and flush
+    # Stop the gpu monitoring thread
     try:
-        _close_logger()
+        gpu_thread.do_run = False
```

Comment:
We only start the gpu_thread if there are GPUs detected in the environment. So the join throw an exception in environments where there are no GPUs. Add a liveness check here: https://docs.python.org/3.7/library/threading.html#threading.Thread.is_alive

I tested this PR and it throws me: 

Exception occurred while stopping GPU monitoring thread: cannot join thread before it is started